valhalla/sif/edgelabel.h

```diff
@@ -72,19 +73,21 @@ class EdgeLabel {
             const bool closure_pruning,
             const bool has_measured_speed,
             const InternalTurn internal_turn,
-            const uint8_t path_id = 0)
+            const uint8_t path_id = 0,
+            const bool destonly = false)
       : predecessor_(predecessor), path_distance_(path_distance), restrictions_(edge->restrictions()),
         edgeid_(edgeid), opp_index_(edge->opp_index()), opp_local_idx_(edge->opp_local_idx()),
         mode_(static_cast<uint32_t>(mode)), endnode_(edge->endnode()),
         use_(static_cast<uint32_t>(edge->use())),
         classification_(static_cast<uint32_t>(edge->classification())), shortcut_(edge->shortcut()),
-        dest_only_(edge->destonly()), origin_(0), destination_(0), toll_(edge->toll()),
-        not_thru_(edge->not_thru()), deadend_(edge->deadend()),
+        origin_(0), destination_(0), toll_(edge->toll()), not_thru_(edge->not_thru()),
+        deadend_(edge->deadend()),
         on_complex_rest_(edge->part_of_complex_restriction() || edge->start_restriction() ||
                          edge->end_restriction()),
         closure_pruning_(closure_pruning), path_id_(path_id), restriction_idx_(restriction_idx),
         internal_turn_(static_cast<uint8_t>(internal_turn)), unpaved_(edge->unpaved()),
         has_measured_speed_(has_measured_speed), cost_(cost), sortcost_(sortcost) {
+    dest_only_ = destonly ? destonly : edge->destonly();
```

Comment:
I took this out of the initializer list so no EdgeLabel usage has to be updated unless it's interested in using HGV destonly